PilotDataPlatform/metadata
# PILOT-932 Archive folders recursively

## Summary

- Change trash/restore behaviour to act recursively if the specified item is a folder.
- Update all unit tests to remove dependencies between tests.

## JIRA Issues

https://indocconsortium.atlassian.net/browse/PILOT-932

## Type of Change


## Testing

Are there any new or updated tests to validate the changes?


## Test Directions

1. Use the post endpoint to create a series of nested items.
2. Trash and restore the parent. All children should be acted upon along with the parent.
